app/assets/javascripts/discourse/app/components/modal/second-factor-edit-security-key.js

```diff
@@ -0,0 +1,34 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+import { inject as service } from "@ember/service";
+
+export default class SecondFactorEditSecurityKey extends Component {
+  @service modal;
+
+  @tracked loading = false;
+
+  @action
+  editSecurityKey() {
+    this.loading = true;
+    this.args.model.user
+      .updateSecurityKey(
+        this.args.model.securityKey.id,
+        this.args.model.securityKey.name,
+        false
+      )
+      .then((response) => {
+        if (response.error) {
+          return;
+        }
+        this.args.model.markDirty();
+      })
+      .catch((error) => {
+        this.args.model.onError(error);
+      })
+      .finally(() => {
+        this.loading = false;
+        this.modal.close();
```

Comment:
```suggestion
        this.args.closeModal();
```
(and remove service injection)